codegen/src/main/java/com/paypal/selion/plugins/CodeGeneratorMojo.java

```diff
@@ -165,9 +162,14 @@ public void execute() throws MojoExecutionException {
                     logger.info("Generating java file for YAML file [" + dataFile.getName() + "] in domain ["
                             + domain + "]");
                     
-                    String tempPackage = basePackage;
+                    String tempPackage = "";
+                    // add the basePackage, iff it contains a value
+                    if (!basePackage.isEmpty()) {
+                        tempPackage += basePackage;
```

Comment:
Good point. Will fix this and merge.

Sent from my Phone
________________________________
From: Sherif Ebady<mailto:notifications@github.com>
Sent: ý4/ý7/ý2015 11:09 AM
To: paypal/SeLion<mailto:SeLion@noreply.github.com>
Cc: Simmons, Doug<mailto:dsimmons@paypal.com>
Subject: Re: [SeLion] Address possible issue in codegen if basePackage is empty (#88)


In codegen/src/main/java/com/paypal/selion/plugins/CodeGeneratorMojo.java<https://github.com/paypal/SeLion/pull/88#discussion_r27904217>:

> @@ -165,9 +162,14 @@ public void execute() throws MojoExecutionException {
>                      logger.info("Generating java file for YAML file [" + dataFile.getName() + "] in domain ["
>                              + domain + "]");
>
> -                    String tempPackage = basePackage;
> +                    String tempPackage = "";
> +                    // add the basePackage, iff it contains a value
> +                    if (!basePackage.isEmpty()) {
> +                        tempPackage += basePackage;


The if test is not strictly necessary now. The direct "tempPackage += basePackage;" now handles both cases.

—
Reply to this email directly or view it on GitHub<https://github.com/paypal/SeLion/pull/88/files#r27904217>.